### PR TITLE
add redaction info to report

### LINKED ISSUE #158: Make redaction approaches clear in the report
We should make the redaction approaches for each output clear in the report. This includes:

- Redaction and rounding thresholds in the event count table.
- Redaction, aggregation and rounding in the top 5 table.
- Redaction and rounding in the figures. Where a sub category is completely redacted, it should be explained why. See https://github.com/opensafely-core/interactive-templates/pull/148#pullrequestreview-1432395612

### PR DESCRIPTION
**Pull request by LFISHER7**
Fixes #158

### CODE REVIEW

File: interactive_templates/templates/v2/analysis/report_template.html [hunk 1]
```diff
@@ -211,7 +216,12 @@ <h2>Most common codes</h2>
                     the specified codelists, only the latest code will contribute to the counts in this table.
                 </p>
                 <table>
-                    <caption><strong>Table 2</strong>. The top 5 most common codes recorded for codelist 1 ({{ codelist_1_name }}).</caption>
+                    <caption><strong>Table 2</strong>. The top 5 most common codes recorded for codelist 1 ({{ codelist_1_name }}).
+                    <strong>Note:</strong>
+                        Codes with a count less than are suppressed to reduce the risk of disclosure of patient identifiable information.
```
Comment (HelenCEBM) [inline interactive_templates/templates/v2/analysis/report_template.html:221]: less than what?

File: interactive_templates/templates/v2/analysis/report_template.html [hunk 2]
```diff
@@ -211,7 +216,12 @@ <h2>Most common codes</h2>
...
+                        They are then rounded to the nearest 7, before being used to calculate the percetage make-up.
```
Comment (HelenCEBM) [inline interactive_templates/templates/v2/analysis/report_template.html:222]:
```suggestion
                        They are then rounded to the nearest 7, before being used to calculate the percentage make-up.
```

File: interactive_templates/templates/v2/analysis/report_template.html [hunk 3]
```diff
@@ -235,7 +245,11 @@ <h2>Most common codes</h2>
                 </table>
 
                 <table>
-                    <caption><strong>Table 3</strong>. The top 5 most common codes recorded for codelist 2 ({{ codelist_2_name }}).</caption>
+                    <caption><strong>Table 3</strong>. The top 5 most common codes recorded for codelist 2 ({{ codelist_2_name }}).
+                    <strong>Note:</strong>
+                        Codes with a count less than are suppressed to reduce the risk of disclosure of patient identifiable information.
```
Comment (HelenCEBM) [inline interactive_templates/templates/v2/analysis/report_template.html:250]: less than what?

File: interactive_templates/templates/v2/analysis/report_template.html [hunk 4]
```diff
@@ -235,7 +245,11 @@ <h2>Most common codes</h2>
...
+                        They are then rounded to the nearest 7, before being used to calculate the percetage make-up.
```
Comment (HelenCEBM) [inline interactive_templates/templates/v2/analysis/report_template.html:251]:
```suggestion
                        They are then rounded to the nearest 7, before being used to calculate the percentage make-up.
```

**Review by HelenCEBM — CHANGES_REQUESTED** [review]
couple of small fixes required..
Out of context of the results, these read a bit odd e.g. why have we suppressed counts less than 50 in one place but less than 10 in others? And why do we round to nearest 10 generally but nearest 7 in some cases?

**Comment by LFISHER7** [comment]
Thanks. I'll make the redaction consistent and will update here

**Comment by LFISHER7** [comment]
I've updated this so the redaction approach is to redact counts <=10 and then round to nearest 10. The only exception is in the event counts, where counts are rounded to the nearest 100 because they're high level counts so the accuracy is less important

**Review by HelenCEBM — APPROVED** [review]
Looks good. Personally I think it would be neater not to suppress values equal to 10, otherwise when we show a 10 it actually only represents values 11-14, but either way is fine!

**Comment by LFISHER7** [comment]
Good point! I've removed redaction of counts equal to 10.

**Pull request merged**